parser.hpp

```diff
@@ -119,13 +120,15 @@ CONSTEXPR CandyState decode_candy_state(char up_char)
 }
 
 constexpr int candy_size = 3;
-constexpr int candy_type_offset = 1;
-constexpr int row_padding = 5;    // space, pipe, pipe, space, newline
-constexpr int noncandy_rows = 3;  // title, border, border
+constexpr int row_padding = 5;  // space, pipe, pipe, space, newline
 
 constexpr auto parse_board_parameters(auto str)
 {
-    BoardParameters ret{ 0, 0, 0, 0 };
+    BoardParameters ret{ 0, 0, 0, 0, 0 };
+
+    constexpr int noncandy_rows = 2;                       // border, border
```

Comment:
Nice comments here. It helps a lot to understand the reason behind these magic numbers :+1:

Reply:
Just followed existing pattern :-)